e2e-tests/greeter-build-test.yaml

```diff
@@ -0,0 +1,121 @@
+# 'greeter' is shell based 'hello world' with different output based on arg0.
+#
+# When executed with zero arguments, greeter will emit its greeting and exit.
+# When executed with 1 or more arguments it will emit "<greeting>, <argv[1]>"
+# 'greeting' is set to basename(argv[0]), unless basename(argv[0]) is 'greeter',
+# in which case the greeting is set to 'howdy'.
+#
+# The main package installs /usr/bin/greeter.
+# subpackages install symlinks to greeter (aloha, hello).
+#
+# The 'tester-blob' melange variable provides a function
+# 'testrun' that runs test.
+package:
+  name: greeter
+  version: 1.0
+  epoch: 0
+  dependencies:
+    runtime:
+      - dash-binsh
+
+environment:
+  contents:
+    packages:
+      - busybox
+
+pipeline:
+  - name: install greeter
+    runs: |
+      bdir=${{targets.destdir}}/usr/bin
+      mkdir -p "$bdir"
+      cat > "$bdir/greeter" <<"EOF"
+      #!/bin/sh
+      greeting=${0##*/}
+      [ "$greeting" = "greeter" ] && greeting="howdy"
+      echo "$greeting${1:+, $1}"
+      EOF
+      chmod 755 "$bdir/greeter"
+
+subpackages:
+  - name: ${{package.name}}-hello
+    description: standard greeting
+    dependencies:
+      runtime:
+        - ${{package.name}}
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.contextdir}}/usr/bin"
+          ln -s greeter "${{targets.contextdir}}/usr/bin/hello"
+    test:
+      pipeline:
+        - runs: |
+            ${{vars.tester-blob}}
+            testrun hello
+
+  - name: ${{package.name}}-aloha
+    description: hawiaan greeting
+    dependencies:
+      runtime:
+        - ${{package.name}}
+    pipeline:
+      - runs: |
+          mkdir -p "${{targets.contextdir}}/usr/bin"
+          ln -s greeter "${{targets.contextdir}}/usr/bin/aloha"
+    test:
+      pipeline:
+        - runs: |
+            ${{vars.tester-blob}}
+            testrun aloha
+
+finalize:
+  - name: post pipeline
+    runs: |
+      find melange-out/ ! -type d
+      mkdir -p melange-out/${{package.name}}/etc
+      echo "finalized-${{package.name}}" > melange-out/${{package.name}}/etc/finalized
```

Comment:
should we use `${{targets.destdir}}` and `${{targets.outdir}}` here instead of hard-coding `melange-out` ?

Reply:
i didn't know of 'targets.outdir'. yes, that fits.  i think i'd pick still using 'package.name' as the assumption/contract should be that `${{targets.outdir}}/<subpackage-or-package>/` and `destdir` doesn't make that clear.

Reply:
done.